### PR TITLE
07_Phyiscal Distributions fig draft

### PR DESCRIPTION
**Pull request by SimonTopp**
Contains code and figure for day 7, physical distributions. Could definitely fit in other categories too and am happy to move around if we'd prefer it to be elsewhere.  Also happy to get any feedback you have!

### CODE REVIEW

**Comment by cnell-usgs** [comment]
OK great, will look at this shortly. Can you write a couple sentences here about where the data came from and the takeaway message? 

and wow! 
image

**Comment by SimonTopp** [comment]
Totally, I included that stuff in the [07_physical_distributions_stopp/README.md](https://github.com/USGS-VIZLAB/chart-challenge-22/pull/8/files#diff-24398545fb65915856c7b0bbd65c6d1db465d918a2f7ccb439020f8184810ac8).  Copied below as well though:
- Overall messaging. How does the chart connect to the day/category? What is the 1-2 sentence takeaway?

Lake color is a key ecological indicator. Drivers of lake color, such as precipitation, ice duration, and land cover, are often correlated to lake elevation. High elevation lakes tend to have fewer nutrient inputs and limited seasonal light availability, leading to lower levels of phytoplankton growth and subsequently bluer colors. Only recently have we been able to examine continental patterns in key ecological indicators such as lake color. In doing so, we're able to better understand how these essential systems are changing in response to long term patterns in climate and land use practices.

- The data source and variables used. Where can the data be found? Is it from USGS or elsewhere? Did you do any pre-processing?

All data used can be downloaded with the provided code. Key datasets are 1) [LimnoSat-US](https://doi.org/10.5281/zenodo.4139694), which contains all cloud free Landsat observations of U.S. lakes from the HydroLakes database between 1984 and 2020 and 2) elevation data from USGS.

- Tools & libraries used

Stacked topography maps: rayshader and elevatr
Stick figure and annotation: xkcd
Layout: gridExtra, ggplot, and cowplot

File: 07_physical_distributions_stopp/01_limnosat_download.R [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+folder <- 'data_in'
+if (file.exists(folder)){
+  folder <- paste0(folder,'/')
+} else {
+  dir.create(folder)
+  folder <- paste0(folder,'/')
+}
+
+# Incerase out timeout because some of these are large files 
+options(timeout = max(6000, getOption("timeout")))
```
Comment (lindsayplatt) [inline 07_physical_distributions_stopp/01_limnosat_download.R]: I have been attempting to download, but keep getting timeouts. I tried without setting a new timeout and with increasing this to 12,000 but haven't been able to successfully run any of the `download.file()` commands below. Any advice @SimonTopp ??

Comment (SimonTopp) [reply 07_physical_distributions_stopp/01_limnosat_download.R]: Unfortunately I didn't have the foresight to chunk up the dataset into smaller bits for easier downloads when I made it.  It's about ~3.7 gb and I've sporadically gotten stuck with timeouts when trying to download it programmatically.  If it's not working for you I'd suggest just downloading it from [zenodo directly](https://doi.org/10.5281/zenodo.4139694).

Edit: Just re-read and saw that you can't download any of them. I've only had issues with the big file (srCorrected_us_hydrolakes_dp_20200628.feather).  Are you using a mac or a windows? I'll go poke around a bit.

Comment (SimonTopp) [reply 07_physical_distributions_stopp/01_limnosat_download.R]: I think it'll be updated with a v2 in the next ~year or so, at which point we'll give a lot more thought to ease of use! For now it's a little unwieldy.

Comment (SimonTopp) [reply 07_physical_distributions_stopp/01_limnosat_download.R]: Just tried and everything works on my end (mac).  Could be you need to explicitly change the method to `libcurl`?

Comment (lindsayplatt) [reply 07_physical_distributions_stopp/01_limnosat_download.R]: Hmmm yeah, I am on Windows. Will change to `libcurl` and see what happens.

File: 07_physical_distributions_stopp/02_limnosat_viz.R [hunk 2]
```diff
@@ -0,0 +1,334 @@
...
+#raster::plot(elev_rast)
+elev_matrix <- raster_to_matrix(elev_rast)
+elev_matrix[elev_matrix < -500] = NA
+attr(elev_matrix,'crs') <- attr(elev_rast,'crs')
+attr(elev_matrix,'extent') <- attr(elev_rast,'extent')
+
+## Make our base layer
+elev_matrix %>%
+  sphere_shade(texture='desert') %>%
+  add_shadow(ray_shade(elev_matrix,zscale=50),0.3) %>%
```
Comment (lindsayplatt) [inline 07_physical_distributions_stopp/02_limnosat_viz.R:78]: Now stuck here. I get the lovely error

```
Error in hillshade[, , 1] * scales::rescale(shadowmap, c(max_darken, 1)) :                                                    
  non-conformable arrays
```

which [I have asked about before](https://github.com/tylermorganwall/rayshader/issues/46) and don't really understand in this context (all the matrices appear to be the same dimensions - 657x405)

Comment (SimonTopp) [reply 07_physical_distributions_stopp/02_limnosat_viz.R:78]: What version `rayshader` are you using?  I think thats a bug that was fixed more recently than the issue you linked to (https://github.com/tylermorganwall/rayshader/issues/176) , but you may need to do devtools install from github and restart your R session.
`devtools::install_github("tylermorganwall/rayshader")`

**Comment by lindsayplatt** [comment]
FYI, I am working on being able to run this code from start to finish and will have a PR to your fork with some suggested changes to get it to run smoothly. In the meantime (darn you `hillshade ... non-conformable arrays` error), I thought I would share a few visual suggestions:

* In your matrix plot, I think it would be nice if the cells covered by ice are a light grey or something so it is clear they aren't just empty.
* I was thinking about how to translate this chart into a story on Twitter and that made me think that it would be nice if that map about lake counts was a separate chart from the rest. I think of it more as supporting information and I could see that + an bit about the lake count distribution would make a nice threaded tweet as part of the overall story.
* Could the matrix plot y-axis text label be horizontal and maybe at the top (right above 4000)? That will definitely make it more legible.
* I think it would be nice if it was clearer that this is summarizing over 35 years of data

Lastly, just want to confirm the following: 
* There are 42,473 lakes represented in this dataset (`length(unique(ls_elev$Hylak_id))`)
* To make the daily matrix plot, the lakes were grouped into 20 elevation bins representing 200 m each (`unique(daily_elev_color$elev_bin)`) and then the _median_ daily lake color was determined per bin (but only for bins where there was at least 50 lakes)
* To make the lake color map, the lakes were grouped into a hex grid and the _median_ lake color for each hex cell for the full time period (March 1984 to June 2020) was calculated.

**Comment by SimonTopp** [comment]
Thanks for the suggestions!  I'll address in more detail tomorrow, but a quick clarification:

>To make the daily matrix plot, the lakes were grouped into 20 elevation bins representing 200 m each (unique(daily_elev_color$elev_bin)) and then the median daily lake color was determined per bin (but only for bins where there was at least 50 lakes)

It's bins with at least 50 observations over the entire time span regardless of number of lakes.

File: 07_physical_distributions_stopp/02_limnosat_viz.R (hunk 2 above)
Comment (lindsayplatt) [reply 07_physical_distributions_stopp/02_limnosat_viz.R:78]: Good point! I installed `0.24.10` from CRAN today. I see `0.27.4` on github and will try with that version.

File: 07_physical_distributions_stopp/01_limnosat_download.R (hunk 1 above)
Comment (lindsayplatt) [reply 07_physical_distributions_stopp/01_limnosat_download.R]: Success!

File: 07_physical_distributions_stopp/02_limnosat_viz.R (hunk 2 above)
Comment (lindsayplatt) [reply 07_physical_distributions_stopp/02_limnosat_viz.R:78]: Success with 0.27.4!

**Comment by lindsayplatt** [comment]
OK, now that you have confirmed what is on the chart, I have a couple of text suggestions: 1) can the blue to green legend say `Median lake color over 36 years`, and 2) can the subtitle say `Distribution of over 42k lakes and their color for 36 years in relation to topography of the Contiguous U.S.`? Just to put a bit more context into it?

**Comment by SimonTopp** [comment]
Ok, just addressed some of your suggestions:
- Moved y label to top of axis
- Adjust legend and title/subtitle text to include # of lakes and # of years
- Made a grey background for ice periods

I didn't move the count plot out because its more time consuming to adjust the layout and whatnot, one thought is if we want to make it a story in the tweet is to use the figure from the manuscript that has more info on the number of Landsat observations we have.
Fig2_ObsDistribution

**Review by cnell-usgs — APPROVED** [review]

**Pull request merged**